### PR TITLE
Add mixins to all commands to configure address and cluster name

### LINKED ISSUE #2045: Add mixins to all commands to configure address and cluster name
Currently the syntax is like `jet -a <address> -n <name> submit <jar> ..` which looks rather odd. We should allow adding address/name after the commands as well.

There might be a few conflicts because `-n` in submit command is already used for other parameters.

### PR DESCRIPTION
**Pull request by caioguedes**
Because the options are already used in other commands, `-n/--cluster-name` and `-a/--addresses` will be replaced by `-t/--targets`. It's possible provide `-t/--targets` option before and after the command, option after command takes precedence.

**Before**
`jet -n jet -a 127.0.0.1:5701 submit job.jar`
**After**
`jet submit -t jet@127.0.0.1:5701 job.jar`
`jet -t jet@127.0.0.1:5701 submit job.jar`

Checklist
- [x] Tags Set
- [x] Milestone Set
- [x] Any breaking changes are documented

Fixes #2045

List of breaking changes:

* `-n/--cluster-name` and `-a/--addresses` are removed from Jet Command Line, and replaced by `-t/--targets`

### CODE REVIEW

**Comment by cangencer** [comment]
thanks a lot for your contribution @caioguedes! Will review this and get back to you promptly.

**Comment by cangencer** [comment]
verify

**Comment by eminn** [comment]
Thanks for the contribution @caioguedes!. One thing needs to be done before merging this pull request is to update the documentation and tutorials which are using the old style addresses.

File: hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java [hunk 1]
```diff
@@ -96,30 +99,14 @@
...
-    @Option(names = {"-a", "--addresses"},
-            split = ",",
-            arity = "1..*",
-            paramLabel = "<hostname>:<port>",
-            description = "Optional comma-separated list of Jet node addresses in the format" +
-                    " <hostname>:<port>, if you want to connect to a cluster other than the" +
-                    " one configured in the configuration file",
-            order = 1
-    )
-    private List<String> addresses;
```
Comment (cangencer) [inline hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java:113]: I think we still need to keep these, as this is a breaking change. You can mark them as deprecated

**Comment by cangencer** [comment]
@caioguedes could you also sign the CLA?

**Comment by caioguedes** [comment]
@cangencer I already signed before, @Holmistr and @mtopolnik have it.

**Comment by caioguedes** [comment]
@eminn I couldn't find mentions about the options in Jet repository... Should I look somewhere else?

**Comment by cangencer** [comment]
I think the one here is needed for CLA assistant to work: https://cla.hazelcast.org/hazelcast/hazelcast-jet?pullRequest=2276 @Holmistr  is there a way to add @caioguedes to the whitelist?

**Comment by cangencer** [comment]
verify

**Comment by caioguedes** [comment]
@eminn, I just updated `site/docs` references, somethings else to update? I am not such about tutoriais they have target version...

**Comment by frant-hartm** [comment]
@caioguedes 
Thank you for the update. :+1: 

Could you please:
- Merge `master` branch into the pr. There are some updates in docs regarding docker, which use the addresses flag which need to be updated as well
- make the cluster name in `--targets` optional, should default to `jet`, this covers majority of uses and makes it more convenient for users
- add more descriptive names for the test methods

We have an upcoming release by end of next week, let us know if you are able to update the PR before that. Thanks!

File: hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java [hunk 2]
```diff
@@ -555,6 +590,59 @@ void merge(Verbosity other) {
         }
     }
 
+    public static class TargetsMixin {
+
+        @Option(names = {"-t", "--targets"},
+                description = "The cluster name and addresses to use if you want to connect to a "
+                    + "cluster other than the one configured in the configuration file.",
+                paramLabel = "<cluster-name>@<hostname>:<port>[,<hostname>:<port>]",
```
Comment (frant-hartm) [inline hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java:598]: Make `cluster-name` optional

File: hazelcast-jet-all/src/test/java/com/hazelcast/jet/server/JetCommandLineTest.java [hunk 3]
```diff
@@ -483,6 +484,72 @@ public void test_submit_argsPassing() {
...
+
+        testTargetsCommand("submit", "-t", target, testJobJarFile.toString());
+        testTargetsCommand("-t", target, "submit", testJobJarFile.toString());
+
+        testTargetsCommand("suspend", "jobName", "-t", target);
+        testTargetsCommand("-t", target, "suspend", "jobName");
+    }
+
+    @Test
+    public void testTargetsPrecedence() {
```
Comment (frant-hartm) [inline hazelcast-jet-all/src/test/java/com/hazelcast/jet/server/JetCommandLineTest.java:523]: Use more descriptive name. It is not immediately obvious if the option before or after command takes precedence.

File: hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java [hunk 4]
```diff
@@ -105,22 +109,25 @@
...
     private List<String> addresses;
 
     @Option(names = {"-n", "--cluster-name"},
-            description = "The cluster name to use when connecting to the cluster " +
-                    "specified by the <addresses> parameter. ",
-            defaultValue = "jet",
-            showDefaultValue = Visibility.ALWAYS,
-            order = 2
+        description = "[DEPRECATED] The cluster name to use when connecting to the cluster " +
+            "specified by the <addresses> parameter. ",
```
Comment (cangencer) [inline hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java:121]: should point to the replaced command

**Comment by caioguedes** [comment]
@frant-hartm, I realized that the default value is not working as expected. Let me fix really quick.

**Comment by frant-hartm** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3006/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3006/jet-oss-pr-builder-3006.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:checkstyle (default) on project hazelcast-jet: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java:72:8: Unused import - picocli.CommandLine.TypeConversionException. [UnusedImports]
--------------------------
</details>
</pre>

**Comment by frant-hartm** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by frant-hartm** [comment]
@caioguedes Thank you very much for your contribution!

**Pull request merged**